#   Análise de dados eleitorado 2020 por municípios
Fonte -> https://www.tse.jus.br/eleicoes/estatisticas/estatisticas-eleitorais

Arquivo -> eleitorado_municipio_2020.csv

In [22]:
#   Importando bibliotecas e criando o DataFrame
import pandas as pd
df = pd.read_csv('eleitorado_municipio_2020.csv', encoding='cp1252', sep=';')

In [23]:
#   Visualizando o DataFrame
df.head().T

0             1          2       3  \
NR_ANO_ELEICAO                         2020          2020       2020    2020   
CD_PAIS                                   1             1          1       1   
NM_PAIS                              Brasil        Brasil     Brasil  Brasil   
SG_REGIAO                                 N             N          N       N   
NM_REGIAO                             Norte         Norte      Norte   Norte   
SG_UF                                    AC            AC         AC      AC   
NM_UF                                  Acre          Acre       Acre    Acre   
CD_MUNICIPIO                           1120          1570       1058    1007   
NM_MUNICIPIO                     ACRELÂNDIA  ASSIS BRASIL  BRASILÉIA  BUJARI   
QTD_ELEITORES                          9169          5913      16844    9613   
QTD_ELEITORES_FEMININO                 4502          2993       8653    4736   
QTD_ELEITORES_MASCULINO                4667          2920       8191    4877   
QTD_ELEITORES_NAOINFORMADO                0             0          0       0   
QTD_ELEITORES_MENOR16                     0             0          0       0   
QTD_ELEITORES_16                         22            21         32      34   
QTD_ELEITORES_17                         85            50        126     106   
QTD_ELEITORES_18                        115            84        233     162   
QTD_ELEITORES_19                        173           122        393     217   
QTD_ELEITORES_20                        255           151        434     256   
QTD_ELEITORES_21A24                     934           608       1789    1085   
QTD_ELEITORES_25A29                    1080           775       2069    1101   
QTD_ELEITORES_30A34                     971           742       1936    1054   
QTD_ELEITORES_35A39                    1082           682       1841    1042   
QTD_ELEITORES_40A44                     962           614       1723    1074   
QTD_ELEITORES_45A49                     761           508       1471     824   
QTD_ELEITORES_50A54                     700           392       1317     711   
QTD_ELEITORES_55A59                     610           324       1048     592   
QTD_ELEITORES_60A64                     493           261        791     471   
QTD_ELEITORES_65A69                     367           199        658     393   
QTD_ELEITORES_70A74                     272           168        467     221   
QTD_ELEITORES_75A79                     152           104        261     150   
QTD_ELEITORES_80A84                      83            54        148      80   
QTD_ELEITORES_85A89                      41            34         73      23   
QTD_ELEITORES_90A94                      10            15         22      13   
QTD_ELEITORES_95A99                       1             4          5       4   
QTD_ELEITORES_MAIORIGUAL100               0             1          7       0   
QTD_ELEITORES_IDADEINVALIDO               0             0          0       0   
QTD_ELEITORES_IDADENAOSEAPLICA            0             0          0       0   
QTD_ELEITORES_IDADENAOINFORMADA           0             0          0       0   
QTD_ELEITORES_COMBIOMETRIA             9101          5862      16638    9576   
QTD_ELEITORES_SEMBIOMETRIA               68            51        206      37   
QTD_ELEITORES_DEFICIENTE                 39            16        126      25   
QUANTITATIVO_NOMESOCIAL                   0             0          0       0   

                                        4  
NR_ANO_ELEICAO                       2020  
CD_PAIS                                 1  
NM_PAIS                            Brasil  
SG_REGIAO                               N  
NM_REGIAO                           Norte  
SG_UF                                  AC  
NM_UF                                Acre  
CD_MUNICIPIO                         1015  
NM_MUNICIPIO                     CAPIXABA  
QTD_ELEITORES                        7540  
QTD_ELEITORES_FEMININO               

In [24]:
#   Visualizando tipos primitivos e se existem valores nulos
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5568 entries, 0 to 5567
Data columns (total 43 columns):
 #   Column                           Non-Null Count  Dtype 
---  ------                           --------------  ----- 
 0   NR_ANO_ELEICAO                   5568 non-null   int64 
 1   CD_PAIS                          5568 non-null   int64 
 2   NM_PAIS                          5568 non-null   object
 3   SG_REGIAO                        5568 non-null   object
 4   NM_REGIAO                        5568 non-null   object
 5   SG_UF                            5568 non-null   object
 6   NM_UF                            5568 non-null   object
 7   CD_MUNICIPIO                     5568 non-null   int64 
 8   NM_MUNICIPIO                     5568 non-null   object
 9   QTD_ELEITORES                    5568 non-null   int64 
 10  QTD_ELEITORES_FEMININO           5568 non-null   int64 
 11  QTD_ELEITORES_MASCULINO          5568 non-null   int64 
 12  QTD_ELEITORES_NAOINFORMADO       5

In [25]:
display(df['NM_MUNICIPIO'])

0         ACRELÂNDIA
1       ASSIS BRASIL
2          BRASILÉIA
3             BUJARI
4           CAPIXABA
            ...     
5563      TOCANTÍNIA
5564        TUPIRAMA
5565      TUPIRATINS
5566    WANDERLÂNDIA
5567         XAMBIOÁ
Name: NM_MUNICIPIO, Length: 5568, dtype: object

#   Verificamos a existência de 5568 municípios, que é o número total de municípios brasileiros, segundo o IBGE 2021.
fonte -> https://www.ibge.gov.br/geociencias/organizacao-do-territorio/estrutura-territorial/15761-areas-dos-municipios.html#:~:text=Sobre%20a%20publica%C3%A7%C3%A3o%20%2D%202021&text=Introdu%C3%A7%C3%A3o-,As%20estimativas%20da%20popula%C3%A7%C3%A3o%20residente%20para%20os%205.568%20munic%C3%ADpios%2C%20mais,Pol%C3%ADtico%E2%80%90Administrativa%20Brasileira%20%E2%80%93%20DPA.
#   Logo o DataFrame está completo em relação a coluna dos municípios

In [26]:
#   Quais são as 5 cidades com maior número total de eleitores deficientes?
df.nlargest(5, 'QTD_ELEITORES_DEFICIENTE')

NR_ANO_ELEICAO  CD_PAIS NM_PAIS SG_REGIAO NM_REGIAO SG_UF  \
5356            2020        1  Brasil        SE   Sudeste    SP   
3658            2020        1  Brasil        SE   Sudeste    RJ   
4988            2020        1  Brasil        SE   Sudeste    SP   
535             2020        1  Brasil        NE  Nordeste    BA   
4008            2020        1  Brasil         S       Sul    RS   

                  NM_UF  CD_MUNICIPIO    NM_MUNICIPIO  QTD_ELEITORES  ...  \
5356          São Paulo         71072       SÃO PAULO        8986687  ...   
3658     Rio de Janeiro         60011  RIO DE JANEIRO        4851887  ...   
4988          São Paulo         64777       GUARULHOS         872880  ...   
535               Bahia         38490        SALVADOR        1897098  ...   
4008  Rio Grande do Sul         85995   CAXIAS DO SUL         333696  ...   

      QTD_ELEITORES_90A94  QTD_ELEITORES_95A99  QTD_ELEITORES_MAIORIGUAL100  \
5356                75281                51741                         5889   
3658                76888                60445                         3989   
4988                  224                   28                            3   
535                  1613                  302                           54   
4008                  531                   98                            9   

      QTD_ELEITORES_IDADEINVALIDO  QTD_ELEITORES_IDADENAOSEAPLICA  \
5356                           77                               0   
3658                           51                               0   
4988                            8                               0   
535                             7                               0   
4008                            0                               0   

      QTD_ELEITORES_IDADENAOINFORMADA  QTD_ELEITORES_COMBIOMETRIA  \
5356                                0                     5404466   
3658                                0                     2495709   
4988                                0                      870016   
535                                 0                     1891978   
4008                                0                      332752   

      QTD_ELEITORES_SEMBIOMETRIA  QTD_ELEITORES_DEFICIENTE  \
5356                     3582221                    154981   
3658                     2356178                     27263   
4988                        2864                     22659   
535                         5120                     19486   
4008                         944                     17884   

      QUANTITATIVO_NOMESOCIAL  
5356                      982  
3658                      283  
4988                       97  
535                       135  
4008                       22  

[5 rows x 43 columns]

In [27]:
#   Calculando a relação entre eleitores deficientes e quantidade de eleitores de cada município,
#   para assim obter a porcentagem de eleitores deficientes por município.
#   Adcionando uma coluna chamada 'PCD%' para guardar esta informação.
pd.set_option('display.precision', 2)
df.insert(0, 'PCD%', df['QTD_ELEITORES_DEFICIENTE']/df['QTD_ELEITORES']*100)
df.head(10).T

0             1          2       3  \
PCD%                                   0.43          0.27       0.75    0.26   
NR_ANO_ELEICAO                         2020          2020       2020    2020   
CD_PAIS                                   1             1          1       1   
NM_PAIS                              Brasil        Brasil     Brasil  Brasil   
SG_REGIAO                                 N             N          N       N   
NM_REGIAO                             Norte         Norte      Norte   Norte   
SG_UF                                    AC            AC         AC      AC   
NM_UF                                  Acre          Acre       Acre    Acre   
CD_MUNICIPIO                           1120          1570       1058    1007   
NM_MUNICIPIO                     ACRELÂNDIA  ASSIS BRASIL  BRASILÉIA  BUJARI   
QTD_ELEITORES                          9169          5913      16844    9613   
QTD_ELEITORES_FEMININO                 4502          2993       8653    4736   
QTD_ELEITORES_MASCULINO                4667          2920       8191    4877   
QTD_ELEITORES_NAOINFORMADO                0             0          0       0   
QTD_ELEITORES_MENOR16                     0             0          0       0   
QTD_ELEITORES_16                         22            21         32      34   
QTD_ELEITORES_17                         85            50        126     106   
QTD_ELEITORES_18                        115            84        233     162   
QTD_ELEITORES_19                        173           122        393     217   
QTD_ELEITORES_20                        255           151        434     256   
QTD_ELEITORES_21A24                     934           608       1789    1085   
QTD_ELEITORES_25A29                    1080           775       2069    1101   
QTD_ELEITORES_30A34                     971           742       1936    1054   
QTD_ELEITORES_35A39                    1082           682       1841    1042   
QTD_ELEITORES_40A44                     962           614       1723    1074   
QTD_ELEITORES_45A49                     761           508       1471     824   
QTD_ELEITORES_50A54                     700           392       1317     711   
QTD_ELEITORES_55A59                     610           324       1048     592   
QTD_ELEITORES_60A64                     493           261        791     471   
QTD_ELEITORES_65A69                     367           199        658     393   
QTD_ELEITORES_70A74                     272           168        467     221   
QTD_ELEITORES_75A79                     152           104        261     150   
QTD_ELEITORES_80A84                      83            54        148      80   
QTD_ELEITORES_85A89                      41            34         73      23   
QTD_ELEITORES_90A94                      10            15         22      13   
QTD_ELEITORES_95A99                       1             4          5       4   
QTD_ELEITORES_MAIORIGUAL100               0             1          7       0   
QTD_ELEITORES_IDADEINVALIDO               0             0          0       0   
QTD_ELEITORES_IDADENAOSEAPLICA            0             0          0       0   
QTD_ELEITORES_IDADENAOINFORMADA           0             0          0       0   
QTD_ELEITORES_COMBIOMETRIA             9101          5862      16638    9576   
QTD_ELEITORES_SEMBIOMETRIA               68            51        206      37   
QTD_ELEITORES_DEFICIENTE                 39            16        126      25   
QUANTITATIVO_NOMESOCIAL                   0             0          0       0   

                                        4                5               6  \
PCD%                                 1.14             0.25            0.83   
NR_ANO_ELEICAO                       2020             2020            2020   
CD_PAIS                                 1                1               1   
NM_PAIS                            Brasil           Brasil          Brasil   
SG_REGIAO                               N          

In [28]:
#   Listando as 33 cidades com maior porcentagem de eleitores PCD
df.nlargest(33, 'PCD%')

PCD%  NR_ANO_ELEICAO  CD_PAIS NM_PAIS SG_REGIAO NM_REGIAO SG_UF  \
3573  21.91            2020        1  Brasil         S       Sul    PR   
3332  17.74            2020        1  Brasil         S       Sul    PR   
3261  15.79            2020        1  Brasil         S       Sul    PR   
658    8.99            2020        1  Brasil        NE  Nordeste    CE   
5149   7.89            2020        1  Brasil        SE   Sudeste    SP   
5350   7.70            2020        1  Brasil        SE   Sudeste    SP   
760    5.76            2020        1  Brasil        NE  Nordeste    CE   
3723   5.72            2020        1  Brasil        NE  Nordeste    RN   
5448   5.71            2020        1  Brasil         N     Norte    TO   
3772   5.68            2020        1  Brasil        NE  Nordeste    RN   
5072   5.48            2020        1  Brasil        SE   Sudeste    SP   
4008   5.36            2020        1  Brasil         S       Sul    RS   
3250   5.29            2020        1  Brasil         S       Sul    PR   
3842   5.22            2020        1  Brasil        NE  Nordeste    RN   
747    5.19            2020        1  Brasil        NE  Nordeste    CE   
3764   5.03            2020        1  Brasil        NE  Nordeste    RN   
3270   4.93            2020        1  Brasil         S       Sul    PR   
5341   4.75            2020        1  Brasil        SE   Sudeste    SP   
5192   4.60            2020        1  Brasil        SE   Sudeste    SP   
3493   4.58            2020        1  Brasil         S       Sul    PR   
5268   4.57            2020        1  Brasil        SE   Sudeste    SP   
5144   4.47            2020        1  Brasil        SE   Sudeste    SP   
3292   4.44            2020        1  Brasil         S       Sul    PR   
5014   4.34            2020        1  Brasil        SE   Sudeste    SP   
3807   4.30            2020        1  Brasil        NE  Nordeste    RN   
5174   4.26            2020        1  Brasil        SE   Sudeste    SP   
3528   4.17            2020        1  Brasil         S       Sul    PR   
5246   4.14            2020        1  Brasil        SE   Sudeste    SP   
661    4.08            2020        1  Brasil        NE  Nordeste    CE   
2502   4.04            2020        1  Brasil         N     Norte    PA   
3893   4.00            2020        1  Brasil         N     Norte    RO   
4880   3.97            2020        1  Brasil        SE   Sudeste    SP   
1151   3.84            2020        1  Brasil        NE  Nordeste    MA   

                    NM_UF  CD_MUNICIPIO              NM_MUNICIPIO  ...  \
3573               Paraná         79871     TRÊS BARRAS DO PARANÁ  ...   
3332               Paraná         74020                     IBEMA  ...   
3261               Paraná         74977                CATANDUVAS  ...   
658                 Ceará         13676                 CARNAUBAL  ...   
5149            São Paulo         62367             NOVA CASTILHO  ...   
5350            São Paulo         61883       SÃO JOÃO DE IRACEMA  ...   
760                 Ceará         15199           PIQUET CARNEIRO  ...   
3723  Rio Grande do Norte         16144          FERNANDO PEDROZA  ...   
5448             Tocatins         92592              BABAÇULÂNDIA  ...   
3772  Rio Grande do Norte         17671     OLHO D'ÁGUA DO BORGES  ...   
5072            São Paulo         66214            JUNQUEIRÓPOLIS  ...   
4008    Rio Grande do Sul         85995             CAXIAS DO SUL  ...   
3250               Paraná         74810               CAMPO LARGO  ...   
3842  Rio Grande do Norte         18872                  UMARIZAL  ...   
747                 Ceará         14974                    PACOTI  ...   
3764  Rio Grande do Norte         17213           MESSIAS TARGINO  ...   
3270               Paraná         75175              CONGONHINHAS  ...   
5341            São Paulo         70815             SÃO FRANCISCO  ...   
5192            São Paulo         68373                PEDRA BELA  ...   
3493      

In [29]:
#   Análisando coluna PCD
df['PCD%'].describe()

count    5568.00
mean        0.60
std         0.75
min         0.00
25%         0.25
50%         0.42
75%         0.69
max        21.91
Name: PCD%, dtype: float64

#   Aqui encontramos uma anomalia:
-> Nos dois blocos supra-exibidos, podemos verificar que a média de % de eleitores PCD é de 0.6% e em 75% dos muniícipios ela não chega a 0.7%.
-> Nota-se também que nas 3 cidades de valores de topo: Três Barras do Paraná, IBEMA, e Catanduva a porcentagem de eleitores PCD destoa até memo do quarto maior valor.
observemos abaixo: 

In [33]:
#   Listando 10 cidade com maior número de eleitores PCD.
df.nlargest(10, 'PCD%')

PCD%  NR_ANO_ELEICAO  CD_PAIS NM_PAIS SG_REGIAO NM_REGIAO SG_UF  \
3573  21.91            2020        1  Brasil         S       Sul    PR   
3332  17.74            2020        1  Brasil         S       Sul    PR   
3261  15.79            2020        1  Brasil         S       Sul    PR   
658    8.99            2020        1  Brasil        NE  Nordeste    CE   
5149   7.89            2020        1  Brasil        SE   Sudeste    SP   
5350   7.70            2020        1  Brasil        SE   Sudeste    SP   
760    5.76            2020        1  Brasil        NE  Nordeste    CE   
3723   5.72            2020        1  Brasil        NE  Nordeste    RN   
5448   5.71            2020        1  Brasil         N     Norte    TO   
3772   5.68            2020        1  Brasil        NE  Nordeste    RN   

                    NM_UF  CD_MUNICIPIO           NM_MUNICIPIO  ...  \
3573               Paraná         79871  TRÊS BARRAS DO PARANÁ  ...   
3332               Paraná         74020                  IBEMA  ...   
3261               Paraná         74977             CATANDUVAS  ...   
658                 Ceará         13676              CARNAUBAL  ...   
5149            São Paulo         62367          NOVA CASTILHO  ...   
5350            São Paulo         61883    SÃO JOÃO DE IRACEMA  ...   
760                 Ceará         15199        PIQUET CARNEIRO  ...   
3723  Rio Grande do Norte         16144       FERNANDO PEDROZA  ...   
5448             Tocatins         92592           BABAÇULÂNDIA  ...   
3772  Rio Grande do Norte         17671  OLHO D'ÁGUA DO BORGES  ...   

      QTD_ELEITORES_90A94  QTD_ELEITORES_95A99  QTD_ELEITORES_MAIORIGUAL100  \
3573                   29                    6                            3   
3332                   15                    2                            1   
3261                   28                    6                            7   
658                    72                   22                            7   
5149                    0                    0                            0   
5350                    0                    0                            0   
760                    54                   12                            2   
3723                   10                    3                            0   
5448                   23                    4                            1   
3772                   18                    2                            3   

      QTD_ELEITORES_IDADEINVALIDO  QTD_ELEITORES_IDADENAOSEAPLICA  \
3573                            0                               0   
3332                            0                               0   
3261                            0                               0   
658                             0                               0   
5149                            0                               0   
5350                            0                               0   
760                             1                               0   
3723                            0                               0   
5448                            0                               0   
3772                            0                               0   

      QTD_ELEITORES_IDADENAOINFORMADA  QTD_ELEITORES_COMBIOMETRIA  \
3573                                0                        8607   
3332                                0                        4506   
3261                                0                        6638   
658                                 0                       12734   
5149                                0                        1195   
5350                                0                        1600   
760                                 0                       10608   
3723                                0                        3019   
5448                                0                        6103   
3772                                0                        4180   

      QTD_ELEITORES_